### ISSUE TITLE
[a11y]: aria-label is hardcoded as 'pagination' for the PaginationNav component

### ISSUE THREAD

**Issue by vgowrabathuni** [carbon-design-system/carbon#20665]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.78.2

### React version

18.3.1

### Automated testing tool and ruleset

e.g. IBM Equal Access Accessibility Checker - 10February2025

### Description

Currently `aria-label` is hardcoded as 'pagination' for the [PaginationNav](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/PaginationNav/PaginationNav.tsx#L508) component. If we use more than one PaginationNav in the page then it's adding the below accessibility violation

```
"Multiple elements with \"navigation\" role do not have unique labels"
```

### Reproduction/example

https://stackblitz.com/edit/kcijtun1?file=src%2FApp.jsx

### Steps to reproduce

Add more than one PaginationNav component and then scan the page with IBM Accessibility checker chrome extension.

### Suggested Severity

Severity 1 = Must be fixed ASAP. The response must be swift. Someone from the team must drop all current work and be immediately reassigned to address the issue.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
